packages/authentication/src/providers/auth-action.provider.ts

```diff
@@ -49,8 +50,26 @@ export class AuthenticateActionProvider implements Provider<AuthenticateFn> {
       return undefined;
     }
 
-    const userProfile = await strategy.authenticate(request);
-    if (!userProfile) {
+    const authResponse = await strategy.authenticate(request);
+    let userProfile: UserProfile;
+
+    // response from `strategy.authenticate()` could return an object of type UserProfile or RedirectRoute
+    if (RedirectRoute.isRedirectRoute(authResponse)) {
+      const redirectOptions = authResponse;
+      // bind redirection url and status as 'authentication.oauth2.redirectUrl' to context
+      // controller should handle actual redirection
+      this.ctx
+        .bind('authentication.redirect.url')
```

Comment:
Please define constants for such keys in `keys.ts`.

Reply:
+1

Reply:
agreed